### PR TITLE
Issue #33: Implement privacy API

### LINKED ISSUE #33: Implement privacy API
I guess this plugin is a `null_provider`?

### PR DESCRIPTION
**Pull request by phager-at**
Closes #33

### CODE REVIEW

File: rules/disposableemails/lang/en/registrationrule_disposableemails.php [hunk 1]
```diff
@@ -31,4 +31,5 @@
 $string['fallbackfailuremessage'] = 'Your email address cannot be veriefied at the moment';
 $string['plugindescription'] = 'Check if user email is on a list of disposable email domains';
 $string['pluginname'] = 'Disposable emails rule';
+$string['privacy:null_provider:reason'] = 'Disposable emails rule does not store any user related data. Only the domain part of user\'s email address gets compared to a list of disposable email domains and happens transiently during form processing.';
```
Comment (daledavies) [inline rules/disposableemails/lang/en/registrationrule_disposableemails.php]: Just a little suggested change, I think this reads a little easier...

`Disposable emails rule does not store any user related data. The domain part of user email addresses will be compared to a list of disposable email domains, this happens transiently during form processing and no data is transmitted.`

File: rules/hcaptcha/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# HCaptcha registration rule
+
+## Description
+
+hCaptcha registration rule plugin uses hCaptcha for bot detection and fraud protection and to harden Moodle's registration form.
```
Comment (daledavies) [inline rules/hcaptcha/README.md]: Maybe replace `and to harden Moodle's registration form` with `during user registration`?

File: rules/hcaptcha/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Privacy policy addition
+
+It is recommended, to be transparent about HCaptcha usage in the site's privacy policy—even for not logged-in users.
```
Comment (daledavies) [inline rules/hcaptcha/README.md]: Probably don't need the first comma and maybe replace "not logged-in" with "unauthenticated"?

File: rules/hibp/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# HCaptcha registration rule
+
+## Description
+
+Have I been pwned registration (HIBP) rule plugin uses HIBP API to check user's provided password for existence in
+publicly available data dumps.
+
+## Privacy policy addition
+
+It is recommended, to be transparent about HIBP usage in the site's privacy policy—even for not logged-in users.
```
Comment (daledavies) [inline rules/hibp/README.md]: Same here?

File: rules/limitdatetime/lang/en/registrationrule_limitdatetime.php [hunk 5]
```diff
@@ -28,5 +28,6 @@
 
  $string['plugindescription'] = 'Only allow registration between two dates';
 $string['pluginname'] = 'Limit by date';
+$string['privacy:null_provider:reason'] = 'Limit by date rule simply limits registration by date. There\'s no user related data involved, stored of processed by the plugin.';
```
Comment (daledavies) [inline rules/limitdatetime/lang/en/registrationrule_limitdatetime.php]: Maybe... "There is no user related data involved, stored, or processed by the plugin." <- added a comma, corrected a typo etc

**Comment by daledavies** [comment]
@phager-at Just a few tiny grammatical corrections

**Comment by phager-at** [comment]
Thanks for the review and corrections! Those are always welcome. :-)

File: rules/limitdatetime/lang/en/registrationrule_limitdatetime.php (hunk 5 above)
Comment (phager-at) [reply rules/limitdatetime/lang/en/registrationrule_limitdatetime.php]: Applied your suggestion!

File: rules/hibp/README.md (hunk 4 above)
Comment (phager-at) [reply rules/hibp/README.md]: Applied your suggestion!

File: rules/hcaptcha/README.md (hunk 3 above)
Comment (phager-at) [reply rules/hcaptcha/README.md]: Applied your suggestion!

File: rules/hcaptcha/README.md (hunk 2 above)
Comment (phager-at) [reply rules/hcaptcha/README.md]: Applied your suggestion!

File: rules/disposableemails/lang/en/registrationrule_disposableemails.php (hunk 1 above)
Comment (phager-at) [reply rules/disposableemails/lang/en/registrationrule_disposableemails.php]: Applied your suggestion!

**Comment by phager-at** [comment]
Thanks Dale, I've applied all your suggestions!
Do you want to take a second look or should I merge it right away?

**Comment by daledavies** [comment]
Just had a quick look over, feel free to merge. Thanks 😁

**Pull request merged**